SimTracker/TrackerHitAssociation/src/TrackerHitAssociator.cc

```diff
@@ -114,7 +114,7 @@ void TrackerHitAssociator::makeMaps(const edm::Event& theEvent, const TrackerHit
           theEvent.labelsForToken(cfToken, labels);
           std::string trackerContainer(labels.productInstance);
```

Comment:
The creation of this string could be avoided too with `strstr` below
```
if(std::strstr(labels.productInstance, highTag) != NULL) ...
```
(in practice, I guess, `nullptr` would work too, but I'm not sure what the standard says here. cppreference.com uses `NULL` in the context of `strstr`).

Even better would be to do the string comparison only once per cfToken, i.e. move the string comparison out of the loop over `*thisContainerHits`.